Design.md

```diff
@@ -32,71 +32,73 @@ Here are the IELE design decisions described from the perspective of differences
+* If the name of the function being called does not correspond to a public function of the contract being called, an exception occurs.
+* In all cases, if a mismatch occurs between the number of arguments or return values of a function and the matching `call` or `ret` instruction, an exception occurs.
+* Because they are no longer needed, we remove the CALLDATA* instructions and the RETURNDATA* instructions.
+* Because of security concerns, we remove the CALLCODE and DELEGATECALL instructions.
 
 ## Arbitrary Precision Words
 
 Unlike EVM, which uses 32-byte unsigned words, IELE has arbitrary-precision signed words.
 
 * Because words are now explicitly signed, the SDIV, SMOD, SLT, and SGT instructions are removed, and DIV, MOD, LT, and GT are replaced with signed versions.
-* An EXPMOD instruction is added to perform exponentiation modulo a particular base, and the MODEXP precompiled contract is removed.
-* SIGNEXTEND is changed to convert an N-byte twos-complement representation of a signed number into its signed value.
-* a TWOS instruction is added to convert a signed integer into an N-byte twos-complement representation of the number.
-* Because integers are arbitrary-precision, the integer operand of BYTE now counts from the least-significant byte instead of the most-significant.
+* An `expmod` instruction is added to perform exponentiation modulo a particular base, and the MODEXP precompiled contract is removed.
+* `sext` (corresponding to EVM's SIGNEXTEND) is changed to convert an N-byte twos-complement representation of a signed number into its signed value.
+* a `twos` instruction is added to convert a signed integer into an N-byte twos-complement representation of the number.
+* Because integers are arbitrary-precision, the integer operand of `byte` (corresponding to EVM's BYTE) now counts from the least-significant byte instead of the most-significant.
 
-## Contract Creation
+## Instructions and Precompiled Contracts
+
+Unlike EVM, where each instruction is just an opcode operating in the values found in the top of the stack, IELE instructions follow a syntax similar to LLVM: each instruction is an opcode followed by some arguments in the form of registers. If the instruction produces a result, it is formed as an assignment where the LHS contains the register(s) to hold the result(s) and the RHS contains the opcode and the arguments. Other changes with respect to EVM include:
```

Comment:
I would replace the two `in` on the first line with `on` and remove the `the` from `the top of`.  Let me formal:
` rule (in => on) ... (in the => on)`

Reply:
Fixed!